src/inmanta/deploy/scheduler.py

```diff
@@ -675,37 +713,89 @@ async def report_resource_state(
+                # Set of dependant resources that are interested in deploys of the current resource
+                concerned_resources: set[ResourceIdStr] = set()
+                provides: Set[ResourceIdStr] = self._state.requires.provides_view().get(resource, set())
+
                 if deployment_result is DeploymentResult.DEPLOYED:
                     self._state.dirty.discard(resource)
+                    is_dirty = False
+
+                    if previous_deployment_result != DeploymentResult.DEPLOYED:
+                        # This resource went from not deployed to deployed
+                        # we have to inform its dependants regardless of event propagation
+                        # so that we can try to re-deploy them in case this resource's previous
+                        # failure was the reason they were skipped
+
+                        # FIXME: stop informing resource skipped because of
+                        #  a custom handler skip via this mechanism
+
+                        dependant_resources: set[ResourceIdStr] = set()
+
+                        for dependant in provides:
+                            if not self._state.resources.get(dependant, None):
+                                continue
+                            if not (dependant_status := self._state.resource_state.get(dependant, None)):
+                                continue
+                            if dependant_status.blocked is BlockedStatus.NO:
+                                dependant_resources.add(dependant)
+                                # Remove timers for unblocked dependant resources because
+                                # they are marked for deployment below.
+                                self._timer_manager.remove_resource(dependant)
+
+                        concerned_resources.update(dependant_resources)
+
                 else:
                     # In most cases it will already be marked as dirty but in rare cases the deploy that just finished might
                     # have been triggered by an event, on a previously successful deployed resource. Either way, a failure
                     # (or skip) causes it to become dirty now.
                     self._state.dirty.add(resource)
+                    is_dirty = True
+
+                    if deployment_result is DeploymentResult.SKIPPED:
+                        # Also add back dependents to the dirty set
+                        # For now we add all dependents
+
+                        dependant_resources = {
+                            dependant for dependant in provides if self._state.resources.get(dependant, None) is not None
+                        }
+                        self._state.dirty.update(dependant_resources)
+                        for dependant in dependant_resources:
+                            # No point in re-trying dependants since this resource was skipped
+                            self._timer_manager.remove_resource(dependant)
```

Comment:
These are not added to concerned_resources?

Reply:
No, since in this case the dependency just finished its deploy and the result was `SKIPPED`, so I don't think we should deploy the dependants ?
Unless they are listening for event propagation, but this is handled below.

Reply:
I agree in concept. This method did grow a lot more than I anticipated, so I'll need a more in-depth review to provide a better response.